addons/website/controllers/main.py

```diff
@@ -309,6 +311,181 @@ def get_dynamic_snippet_templates(self, filter_name=False):
+        """
+        Returns list of results according to the term and options
+
+        Params:
+            search_type (str): search type targeting search method website._autocomplete_%s
+            term (str): search term written by the user
+            order (str)
+            limit (int), default to 5: number of results to consider
+            max_nb_chars (int): max number of characters for text fields
+            options (dict): specific to search_type
+
+        Returns:
+            dict (or False if no result)
+                - 'results' (list): results (only their needed field values)
+                        note: the monetary fields will be strings properly formatted and
+                        already containing the currency
+                - 'results_count' (int): the number of results in the database
+                        that matched the search query
+                - 'parts' (dict): presence of fields across all results
+        """
+        order = self._get_search_order(order)
+        options = options or {}
+        handler = getattr(request.website, '_autocomplete_%s' % search_type, None)
+        if not handler:
+            return {
+                'results': [],
+                'results_count': 0,
+                'parts': {},
+            }
+        mappings = []
+        if search_type == 'all':
+            results = handler(term, limit, order, options)
+            results_count = 0
+            results_data = []
+            for count, data, mapping in results:
+                if data:
+                    results_count += count
+                    # keep track of the mapping index for each source of results
+                    for record in data:
+                        record['_origin'] = len(mappings)
+                    mappings.append(mapping)
+                    results_data += data
+            # Only supported order for 'all' is on name
+            results_data.sort(key=lambda r: r.get('name', ''), reverse='name desc' in order)
+            results_data = results_data[:limit]
+        else:
+            results_count, results_data, mapping = handler(term, limit, order, options)
+            mappings.append(mapping)
+        result = []
+        for record in results_data:
+            mapping = mappings[record.get('_origin', 0)]
+            mapped = {
+                '_fa': record.get('_fa'),
+            }
+            for mapped_name, field_meta in mapping.items():
+                value = record.get(field_meta.get('name'))
+                if value:
+                    field_type = field_meta.get('type')
+                    if field_type == 'text':
+                        if value and len(value) > max_nb_chars:
+                            value = "%s..." % value[:(max_nb_chars - 3)]
+                        if field_meta.get('match') and value and term:
+                            pattern = '(%s)' % '|'.join([re.escape(search_term) for search_term in term.split(' ')])
```

Comment:
In this code `term.split(' ')`, consecutive delimiters are not grouped and empty strings are delimited, that's why you will get a Stacktrace (in `re.split(pattern, value, flags=re.I)`) when you type a space in search input.

`'(%s)' % '|'.join([re.escape(search_term) for search_term in term.split(' ')])` **<=>** `'(%s)' % '|'.join(['', ''])` => returns `'(|)'` as pattern used in `.split()`.